sssd_test_framework/utils/realmd.py

```diff
@@ -0,0 +1,115 @@
+"""Manage enrollment in realms."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from pytest_mh import MultihostHost, MultihostUtility
+from pytest_mh.conn import ProcessResult
+
+__all__ = [
+    "RealmdUtils",
+]
+
+
+class RealmdUtils(MultihostUtility[MultihostHost]):
+    """
+    Call commands from realmd
+
+    .. code-block:: python
+        :caption: Example usage
+
+    @pytest.mark.topology(KnownTopology.AD)
+    def test_realm_discover(client: Client, provider: ADProvider):
+        p_domain = f"{provider.host.domain}"
+        r = client.realmd.discover([p_domain, "--use-ldaps"])
+        assert provider.host.domain in r.stdout, "realm failed to discover domain info"
+    """
+
+    def discover(self, domain: str, args: list[Any] | None = None) -> ProcessResult:
+        """
+        Call ``realm discover``  with given arguments.
+
+        Discover a realm and it's capabilities
+
+        :param domain: Discover information about domains
+        :type domain: str,
+        :param args: additional arguments to the discover operation, defaults to None
+        :type args: list[Any] | None, optional
+        """
+        if args is None:
+            args = []
+
+        return self.host.conn.exec(["realm", "discover", *args])
+
+    def leave(self, args: list[Any] | None = None) -> ProcessResult:
+        """
+        Call ``realm leave`` with given arguments.
+
+        Deconfigure a client system from a realm.
+
+        :param args: additional arguments to pass to the leave operation, defaults to None
```

Comment:
Nitpick, The pattern is, capitalized ```:param: **D**escription blah blah blah**.**```, no period when ```:param: Description, blah blah, default to None```

I want it to be capitalized, the period thing, at least be consistent within the file, but it's easier to conform to the rest of the framework.